lib/ddtrace/contrib/rack/rum_injection.rb

```diff
@@ -0,0 +1,239 @@
+require 'ddtrace/contrib/rack/ext'
+require 'ddtrace/environment'
+require 'delegate'
+require 'date'
+
+module Datadog
+  module Contrib
+    # Rack module includes middlewares that are required to trace any framework
+    # and application built on top of Rack.
+    module Rack
+      # RumInjection ensures that the Rack Response has rum information
+      # injected into it's html. The middleware modifies the response body
+      # of non-cached html so that it can be retrieved by the rum browser-sdk in
+      # the application frontend.
+      # rubocop:disable Metrics/ClassLength
+      class RumInjection
+        include Datadog::Environment::Helpers
+
+        RUM_INJECTION_FLAG = 'datadog.rum_injection_flag'.freeze
```

Comment:
How will RUM_INJECTION_FLAG be used?
I notice we only check for it, but never set it.
Is this an (undocumented) user settings?

Reply:
It gets set on the rack env 

here: https://github.com/DataDog/dd-trace-rb/pull/1139/files/68f0092be377639661137874974daa0594e1ba59#diff-e49d67d2fbda9897a00cada5fc03f70aR87

and we don't inject if it's set

check here: https://github.com/DataDog/dd-trace-rb/pull/1139/files/68f0092be377639661137874974daa0594e1ba59#diff-e49d67d2fbda9897a00cada5fc03f70aR115

The idea is to ensure the middleware is not called twice for any reason or the html is not inserted twice. if it is attempted to be inserted twice for whatever reason, we do not re-inject again.

Eventually, when we have a helper for manual insertion in a template, we can then set this env var to ensure we don't also inject again here.

Reply:
👍